submittal/src/intTest/java/com/kyriba/submittal/api/SpeakerControllerIntTest.java

```diff
@@ -0,0 +1,203 @@
+package com.kyriba.submittal.api;
+
+import com.kyriba.submittal.service.KeyGenerator;
+import com.kyriba.submittal.service.KeyType;
+import org.apache.commons.lang.RandomStringUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.EnableAspectJAutoProxy;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+import static org.mockito.BDDMockito.given;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+
+/**
+ * @author M-ABL
+ */
+@SpringBootTest
+@ExtendWith(SpringExtension.class)
+@EnableAspectJAutoProxy
+@ActiveProfiles("inttest")
+class SpeakerControllerIntTest
+{
+  @Autowired
+  protected WebApplicationContext context;
+
+  private MockMvc mockMvc;
+
+  @MockBean
+  private KeyGenerator keyGenerator;
+
+
+  @BeforeEach
+  void setUp()

```

Comment:
You mixed several approaches here. I will explain testing approaches in the next module.